### PR TITLE
feat: added maximum message size to wire protocol codec

### PR DESCRIPTION
**Pull request by adenh93**

### CODE REVIEW

File: protocol/src/codec.rs [hunk 1]
```diff
@@ -13,8 +15,10 @@ pub struct MessageCodec;
 impl Encoder<Frame> for MessageCodec {
     type Error = anyhow::Error;
 
-    fn encode(&mut self, item: Frame, dst: &mut BytesMut) -> Result<(), Self::Error> {
```
Comment (petehayes102) [inline protocol/src/codec.rs:16]: I'm dubious about removing these. I know it equates to the same thing, but it makes `type Error` meaningless and could obfuscate problems down the track if the error or result types change.

File: protocol/src/codec.rs [hunk 2]
```diff
@@ -56,6 +62,18 @@ impl Decoder for MessageCodec {
     }
 }
 
+fn validate_payload_length(length: u64) -> Result<()> {
+    if length > MAX_MESSAGE_SIZE as u64 {
+        bail!(
```
Comment (petehayes102) [inline protocol/src/codec.rs:67]: Could you just use an `Err` variant here? There's no need for bail I don't think?

**Review by petehayes102 — CHANGES_REQUESTED** [review]
Just nitpicking. Looks great!!

**Comment by adenh93** [comment]
> Just nitpicking. Looks great!!

No worries, they were valid points! 👌

**Pull request merged**